search-engines.lisp

```diff
@@ -1074,6 +1074,130 @@ REQUEST-ARGS is a list of args to pass to request function."
+                             (:long "long")))
+  (category "categoryOneOf" ((:all "")
+                             (:music "1")
+                             (:films "2")
+                             (:vehicles "3")
+                             (:art "4")
+                             (:sports "5")
+                             (:travels "6")
+                             (:gaming "7")
+                             (:people "8")
+                             (:comedy "9")
+                             (:entertainment "10")
+                             (:news-and-politics "11")
+                             (:how-to "12")
+                             (:education "13")
+                             (:activism "14")
+                             (:science-and-technology "15")
+                             (:animals "16")
+                             (:kids "17")
+                             (:food "18")))
+  (licence "licenceOneOf" ((:all "")
+                           (:by "1")
+                           (:by-sa "2")
+                           (:by-nd "3")
+                           (:by-nc "4")
+                           (:by-nc-sa "5")
+                           (:by-nc-nd "6")
+                           (:cc0 "7")))
+  (language "languageOneOf" ((:all "")
+                             (:english "en")
+                             (:francais "fr")
+                             (:japanese "ja")
+                             (:euskara "eu")
+                             (:catala "ca")
+                             (:czech "cs")
+                             (:esperanto "eo")
+                             (:bulgarian "el")
+                             (:deutsch "de")
+                             (:italiano "it")
+                             (:nederlands "nl")
+                             (:espanol "es")
+                             (:occitan "oc")
+                             (:gaelic "gd")
+                             (:chinese "zh")
+                             (:portugues "pt")
+                             (:svenska "sv")
+                             (:polski "pl")
+                             (:suomi "fi")
+                             (:russian "ru")))
+  (host "host" ((:default "")))
+  (tags "tagsAllOf" ((:default ""))))
+
+(define-search-engine sourcehut
+    (:shortcut "sh"
+     :fallback-url (quri:uri "https://sr.ht")
+     :base-search-url "https://sr.ht/projects?search=~a"
+     :documentation "Sourcehut project search `nyxt:search-engine'.")
+  (sort-by "sort" ((:recent "recently-updated")
+                   (:active "longest-active"))))
+
+(define-search-engine libgen
+    (:shortcut "lg"
+     :fallback-url (quri:uri "https://libgen.li")
+     :base-search-url "https://libgen.li/index.php?req=~a"
+     :documentation "Library Genesis `nyxt:search-engine'.")
+  (results "res" ((:default "")
+                  (:twenty-five "25")
+                  (:fifty "50")
+                  (:one-hundred "100")))
```

Comment:
Does is allow an arbitrary integers? Even if not, you better use literal integers as options there, instead of lengthy keywords.

Reply:
Or add integers to the keywords.